### PR TITLE
Adding principle for client hints, fixes #307

### LINKED ISSUE #307: New principle: When to use client hints
This was brought up in our discussion of [client hints reliability mechanism](https://github.com/w3ctag/design-reviews/issues/549) - in breakout A on [week-of April 19](https://github.com/w3ctag/meetings/blob/gh-pages/2021/telcons/04-19-agenda.md). The question was raised : should we have a design principle on when to use / not use client hints as opposed to other mechanisms?

### PR DESCRIPTION
**Pull request by torgo**
***
<a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/pull/434.html" title="Last updated on Apr 21, 2023, 2:28 AM UTC (97cf72d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/434/94996e0...97cf72d.html" title="Last updated on Apr 21, 2023, 2:28 AM UTC (97cf72d)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -2814,6 +2814,20 @@ and make sure your choice of algorithm is proven, and up-to-date.
 Not only do they become obsolete or insecure,
 cryptographic protocols and algorithms also evolve quickly.
 
+<h3 id="client-hints">Client Hints must not be used to expose information that is not already 
+exposed to web pages</h3>
+
+Client hints are an important optimization, but cannot be the sole means by which information
+ is exposed to sites. As it says in
```
Comment (hober) [inline index.bs:2821]:
```suggestion
is exposed to sites. As it says in 
```

File: index.bs [hunk 2]
```diff
@@ -2814,6 +2814,20 @@ and make sure your choice of algorithm is proven, and up-to-date.
 Not only do they become obsolete or insecure,
 cryptographic protocols and algorithms also evolve quickly.
 
+<h3 id="client-hints">Client Hints must not be used to expose information that is not already 
+exposed to web pages</h3>
+
+Client hints are an important optimization, but cannot be the sole means by which information
+ is exposed to sites. As it says in 
+ <cite><a href=https://datatracker.ietf.org/doc/html/rfc8942#section-4.1-5>RFC 8942 §4.1</a></cite>
```
Comment (hober) [inline index.bs:2822]:
```suggestion
<cite><a href=https://datatracker.ietf.org/doc/html/rfc8942#section-4.1-5>RFC 8942 §4.1</a></cite>
```

**Review by hober — APPROVED** [review]
Please fix the bikeshed indentation issues and then this is good to go.

File: index.bs [hunk 3]
```diff
@@ -2814,6 +2814,20 @@ and make sure your choice of algorithm is proven, and up-to-date.
 Not only do they become obsolete or insecure,
 cryptographic protocols and algorithms also evolve quickly.
 
+<h3 id="client-hints">Client Hints must not be used to expose information that is not already
```
Comment (cynthia) [inline index.bs:2817]:
```suggestion
<h3 id="client-hints">Do not expose new information through Client Hints</h3>
```

File: index.bs [hunk 4]
```diff
@@ -2814,6 +2814,20 @@ and make sure your choice of algorithm is proven, and up-to-date.
 Not only do they become obsolete or insecure,
 cryptographic protocols and algorithms also evolve quickly.
 
+<h3 id="client-hints">Client Hints must not be used to expose information that is not already 
+exposed to web pages</h3>
```
Comment (cynthia) [inline index.bs:2818]:
```suggestion
```

File: index.bs [hunk 5]
```diff
@@ -2814,6 +2814,20 @@ and make sure your choice of algorithm is proven, and up-to-date.
...
+
```
Comment (cynthia) [inline index.bs:2819]:
```suggestion

When using Client Hints, don't expose new information that the web page does not already know.

```

**Review by cynthia — CHANGES_REQUESTED** [review]

File: index.bs (hunk 5 above)
Comment (hober) [reply index.bs:2819]: I think this wording is a little funny. How about this?

```suggestion

When using Client Hints, don't expose information that the web page does not already have access to.

```

**Review by cynthia — COMMENTED** [review]
LGTM

**Review by cynthia — APPROVED** [review]

**Review by plinss — APPROVED** [review]

File: index.bs [hunk 6]
```diff
@@ -2814,6 +2814,21 @@ and make sure your choice of algorithm is proven, and up-to-date.
...
+is exposed to sites. As it says in 
+<cite><a href=https://datatracker.ietf.org/doc/html/rfc8942#section-4.1-5>RFC 8942 §4.1</a></cite>
+where client hints are defined:
+
+<blockquote cite="https://datatracker.ietf.org/doc/html/rfc8942#section-4.1-5">
+<p>Therefore, features relying on this document to define Client Hint headers 
+<strong>MUST NOT</strong> provide new information that is otherwise not made available to 
+the application by the user agent, such as existing request headers, HTML, CSS, or JavaScript.
+</blockquote>
+
```
Comment (LeaVerou) [inline index.bs:2831]: Needs concrete guidance. How about something like this? (rough draft)

```suggestion

If you are designing a new client hint that exposes information that is not available to the web page through other means, please consider the security and privacy implications of exposing this information. If there are none, please define an API for accessing this information before defining the client hint.
```

**Pull request merged**